src/android.ts

```diff
@@ -256,6 +258,15 @@ export const androidApi = <RunTarget extends SupportedRunTarget<'android'>>(
         },
     },
 
+    async waitForDevice(tries = 20) {
+        let booted = false;
+        try {
+            booted = await retryCondition(() => this._internal.hasDeviceBooted(true), tries, 100);
```

Comment:
Because it is different where the error is thrown, either in a nested promise or in the function body directly. This changes the error handling behavior and how much context to the error is available. I want the error to occur in the parent promise.

Reply:
Not sure what you mean. Your code and this

```ts
const booted = await retryCondition(
    () => this._internal.hasDeviceBooted({ waitForDevice: true }),
    tries,
    100
).catch((e) => {
    throw new Error('Failed to wait for device: Error in adb', { cause: e });
});
```

produce pretty much the exact same error if `hasDeviceBooted` throws:

```
/home/benni/coding/JS/tweasel/appstraction/src/android.ts:263
            throw new Error('Failed to wait for device: Error in adb', { cause: e });
                  ^


Error: Failed to wait for device: Error in adb
    at Object.waitForDevice (/home/benni/coding/JS/tweasel/appstraction/src/android.ts:263:19)
    at <anonymous> (/home/benni/coding/JS/tweasel/appstraction/examples/android-device.ts:20:5) {
  [cause]: Error: oops, something went wrong
      at Object.hasDeviceBooted (/home/benni/coding/JS/tweasel/appstraction/src/android.ts:148:19)
      at <anonymous> (/home/benni/coding/JS/tweasel/appstraction/src/android.ts:261:64)
      at retryCondition (/home/benni/coding/JS/tweasel/appstraction/src/util.ts:29:20)
      at Object.waitForDevice (/home/benni/coding/JS/tweasel/appstraction/src/android.ts:261:28)
      at <anonymous> (/home/benni/coding/JS/tweasel/appstraction/examples/android-device.ts:20:19)
      at <anonymous> (/home/benni/coding/JS/tweasel/appstraction/examples/android-device.ts:58:1)
      at ModuleJob.run (node:internal/modules/esm/module_job:194:25)
}
```

vs.

```
/home/benni/coding/JS/tweasel/appstraction/src/android.ts:264
            throw new Error('Failed to wait for device: Error in adb', { cause: e });
                  ^


Error: Failed to wait for device: Error in adb
    at <anonymous> (/home/benni/coding/JS/tweasel/appstraction/src/android.ts:264:19)
    at Object.waitForDevice (/home/benni/coding/JS/tweasel/appstraction/src/android.ts:259:24)
    at <anonymous> (/home/benni/coding/JS/tweasel/appstraction/examples/android-device.ts:20:5) {
  [cause]: Error: oops, something went wrong
      at Object.hasDeviceBooted (/home/benni/coding/JS/tweasel/appstraction/src/android.ts:148:19)
      at <anonymous> (/home/benni/coding/JS/tweasel/appstraction/src/android.ts:260:34)
      at retryCondition (/home/benni/coding/JS/tweasel/appstraction/src/util.ts:29:20)
      at Object.waitForDevice (/home/benni/coding/JS/tweasel/appstraction/src/android.ts:259:30)
      at <anonymous> (/home/benni/coding/JS/tweasel/appstraction/examples/android-device.ts:20:19)
      at <anonymous> (/home/benni/coding/JS/tweasel/appstraction/examples/android-device.ts:58:1)
      at ModuleJob.run (node:internal/modules/esm/module_job:194:25)
}
```

The only difference is that my version has an additional `<anonymous>` in the stack trace from the catch function.